### ISSUE TITLE
Installing only devDependencies - `swagger-express-middleware` prevents `glob` from installing

### ISSUE THREAD

**Issue by chloebrett** [APIDevTools/swagger-express-middleware#172]
I am trying to install the `glob` module and have encountered a weird bug I can't seem to make sense of. I'm not sure whether this is an issue with `swagger-express-middleware` or something I'm missing about how NPM works. Potentially it's something else entirely. Either way, I thought someone working on this project might be able to shed some light. Thanks for reading and feel free to direct me elsewhere if there's a better place to ask this (I've already tried StackOverflow, but all I got was confirmation that the issue I found does indeed exist).

This problem is best explained by giving instructions to replicate it:

Package.json (this isn't a snippet, this is the whole file for the purposes of replicating the problem):

```
{
  "devDependencies": {
    "glob": "^7.1.6"
  }
}
```

Bash script:

```
#!/bin/bash -euxo pipefail

rm -rf node_modules || true    # remove folder if present
rm package-lock.json || true   # remove lockfile if present
npm install --only=development # install devDependencies only
ls node_modules/glob           # check whether glob was installed
```

This works as expected - `glob` is installed, the contents of the glob folder are printed to the screen, and everybody's happy.

But when I put `swagger-express-middleware` in the main dependencies section:

Package.json:

```
{
  "dependencies": {
    "swagger-express-middleware": "^4.0.0"
  },
  "devDependencies": {
    "glob": "^7.1.6"
  }
}
```

And run the same bash script as above, then I get the following error:

`ls: node_modules/glob: No such file or directory`

In other words, the `glob` module never got installed, even though the only difference was the presence of a dependency that wasn't even installed either, since we specifically only installed dev dependencies anyway via `npm install --only=development`.

Why might this be the case? I cannot for the life of me figure out why the presence of `swagger-express-middleware` (and no other modules are doing this, mind you) would prevent `glob` from being installed.

Are there any other examples of conflicts like this?

Device details, in case it makes a difference: macOS Catalina 10.15.7, NPM 6.14.8, Node 12.18.2
